### PR TITLE
fix(executor, frontend): `BatchHopWindow` executor should derive window expr in frontend

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Batch Hop executor should derive expr in frontend. e2e test previously panicked with variable duration interval due to not inlining session timezone.

Next step: stream

## Checklist For Contributors

- [X] I have written necessary rustdoc comments
- [X] I have added necessary unit tests and integration tests
- [X] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [X] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [X] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [X] My PR **DOES NOT** contain user-facing changes.

## Related
Fixes part of: https://github.com/risingwavelabs/risingwave/issues/7334

### CODE REVIEW

File: src/batch/src/executor/hop_window.rs [hunk 1]
```diff
@@ -153,116 +170,33 @@ impl HopWindowExecutor {
...
-                output_type.clone(),
-                time_col_ref,
-                window_size_sub_slide_expr,
-            )?,
-            window_slide_expr,
-        )?;
-
-        let mut window_start_exprs = Vec::with_capacity(units);
-        let mut window_end_exprs = Vec::with_capacity(units);
+        let _output_type = DataType::window_of(&time_col_data_type).unwrap();
```
Comment (jon-chuang) [inline src/batch/src/executor/hop_window.rs:173]: remove

**Review by st1page — APPROVED** [review]
generally LGTM.
A potential problem is that it might produce a large plan node. It might be critical in serving situations(with high QPS). But I think the window TVF is hardly used in that situation. So it is not necessary to do that optimization prematurely.

File: src/frontend/src/optimizer/plan_node/logical_hop_window.rs [hunk 2]
```diff
@@ -362,11 +364,106 @@ impl PredicatePushdown for LogicalHopWindow {
     }
 }
 
+pub fn derive_window_start_and_end_exprs(
+    logical_hop: &LogicalHopWindow,
+) -> Result<(Vec<ExprImpl>, Vec<ExprImpl>)> {
```
[the remark this answers is not in the document]
Comment (jon-chuang) [reply src/frontend/src/optimizer/plan_node/logical_hop_window.rs:369]: Indeed, let me do this

**Comment by jon-chuang** [comment]
> It might be critical in serving situations(with high QPS). But I think the window TVF is hardly used in that situation. So it is not necessary to do that optimization prematurely.

Yes, it does not seem too likely to use a hop window in serving scenario.

**Pull request merged**